### ISSUE TITLE
Moving tags for releases

### ISSUE THREAD

**Issue by Noobish1** [DataDog/dd-sdk-ios#1979]
### Question

Hi there, are you occasionally moving tags on releases?

I believe it leads to this SPM error on our on-prem CI runners, which is a pain because we have to nuke the SPM cache in `~/Library/org.swift.swiftpm` on each machine.

> xcodebuild: error: Could not resolve package dependencies:
  Revision df7f98ea7cadbe2f8e775b474573f3db7ba81938 for dd-sdk-ios remoteSourceControl https://github.com/DataDog/dd-sdk-ios.git version 2.15.0 does not match previously recorded value 0331b7ef6605b6c014f169ac96993dfbd450c010

**Comment by ncreated** [comment]
Hey @Noobish1 👋. Indeed, we moved the `2.15.0` tag recently between the two commits mentioned in your CI/CD error. It was done within [**2.15.0** GH release](https://github.com/DataDog/dd-sdk-ios/releases/tag/2.15.0) while it was in "pre-release" stage and not yet marked as "latest" in GitHub.

**The flaw is entirely on us** - we assumed that moving tags within pre-release is safe for dependency manager, but learned hard that it is not. We noticed similar impact in our CI/CD and we recover from it through https://github.com/DataDog/dd-sdk-ios/pull/1980.

We haven't anticipated this may impact other CI/CD systems which is what we apologize for 😞. Related post-mortem was already recorded within our team to ensure this problem won't repeat in the future.

The problem originated in one of our release jobs failing due to missing CI tooling - we recently migrated all automations to GitLab and `2.15.0` is the first release using it. The problem was fixed in #1975, then merged and then we re-started the release pipeline by moving the tag. We wrongly assumed it is safe for GH pre-release. As a follow-up action to this problem, we will refine our Smoke Tests jobs to anticipate failures in release job, so we avoid failures in tags.

---

Mitigation:
- SPM users: as mentioned by @Noobish1, nuke the SPM cache in `~/Library/org.swift.swiftpm`
- Carthage users: cleanup the `dd-sdk-ios` cache under `~/Library/Caches/org.carthage.CarthageKit/dependencies/`
